src/Device/Driver/AR62xx.cpp

```diff
@@ -0,0 +1,496 @@
+  frequency_id &= FREQUENCY_BITMASK;
+
+  /** get channel out of frequency */
+  uint8_t channel = frequency % 100;
+
+  /** encode channel and add to frequency */
+  if (channel <= 15)
+    frequency_id += channel / 5;
+  else if (channel <= 40)
+    frequency_id += channel / 5 - 1;
+  else if (channel <= 65)
+    frequency_id += channel / 5 - 2;
+  else if (channel <= 90)
+    frequency_id += channel / 5 - 3;
+
+  return (frequency_id);
+}
+
+static uint16_t CRCBitwise(uint8_t *data, size_t len)
+{
+  uint16_t crc = 0x0000;
+  for (size_t j = len; j > 0; j--)
+  {
+    crc ^= (uint16_t)(*data++) << 8;
+    for (int i = 0; i < 8; i++)
+    {
+      if (crc & 0x8000)
+        crc = (crc << 1) ^ 0x8005;
+      else
+        crc <<= 1;
+    }
+  }
+  return (crc);
+}
+
+unsigned int AR62xxDevice::GenerateSetStationCommand(uint8_t *command, unsigned int active_passive, const RadioFrequency frequency)
+{
+  /** length of the command */
+  unsigned int command_length = 0;
+
+  /** convert the active-frequency to an AR62xx-ID */
+  uint16_t active_frequency_idx = FrequencyToAR62xxId(active_frequency);
+
+  /** convert the passive-frequency to an AR62xx-ID */
+  uint16_t passive_frequency_idx = FrequencyToAR62xxId(passive_frequency);
+
+  /** command-start */
+  command[command_length++] = 0xA5;
+
+  /** add prot-ID */
+  command[command_length++] = 0x14;
+  command[command_length++] = 5;
+
+  /** set active station */
+  if (active_passive == ACTIVE_STATION)
+    active_frequency_idx = FrequencyToAR62xxId(frequency);
+
+  /** set passive station */
+  else
+    passive_frequency_idx = FrequencyToAR62xxId(frequency);
+
+  /** setting frequencies */
+  command[command_length++] = 22;
+  command[command_length++] = GetNthByte(active_frequency_idx, 1);
+  command[command_length++] = GetNthByte(active_frequency_idx, 0);
+  command[command_length++] = GetNthByte(passive_frequency_idx, 1);
+  command[command_length++] = GetNthByte(passive_frequency_idx, 0);
+
+  /** Creating the binary value */
+  uint16_t command_crc = CRCBitwise(command, command_length);
+  command[command_length++] = GetNthByte(command_crc, 1);
+  command[command_length++] = GetNthByte(command_crc, 0);
+  return command_length;
+}
+
+bool AR62xxDevice::ParseReceivedData(const uint8_t *data, size_t len)
+{
+  size_t idx = 0;
+  uint16_t crc;
+  uint16_t rec_buf_len = 0;
+  int command_length;
+  int rec_bufsize = 127;
+  uint8_t command[rec_bufsize];
+
+  /** loop as long we have data */
+  while (idx < len)
+  {
+    /** check if we received a header*/
+    if (data[idx] == 0xA5)
+      /** reset buffer length */
+      rec_buf_len = 0;
+
+    /** check if the buffer-size is reached */
+    if (rec_buf_len >= rec_bufsize)
+      /** reset buffer length */
+      rec_buf_len = 0;
+
+    /** write command in buffer */
+    command[rec_buf_len++] = (char)data[idx++];
+
+    /** if we received data we have to find the start of the protocol
```

Comment:
This is a doxygen comment, but this is not API documentation.

Reply:
Hi @MaxKellermann, just one question before i change all the comments (even if it can be done by search/replace) ... which commenting-style do you prefer for non-api-documentation?
is `/* sth ... */`ok (also for single line comments) or `// sth ... ` or sth else?

Reply:
All C++ comments without ASCII art (no extra `*` and no ASCII art lines) are acceptable. I personally prefer plain C comments `/* ... */` over C++ one-liners `// ...` but for XCSoar, both are ok.

Reply:
done :-)